### PR TITLE
RUN-3739 auth resource fetch

### PR DESCRIPTION
**Pull request by wenjunche**
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a4d5662bd581266877d663a)
[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a4d56ebbd581266877d663b)

[Javascript PR](https://github.com/openfin/javascript-adapter/pull/373)

### CODE REVIEW

**Comment by wenjunche** [comment]
Tests from the 2nd commit:

[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a537fe4bd581266877d6648)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a537f3dbd581266877d6647)

**Comment by HarsimranSingh** [comment]
@wenjunche can you move the jira ticket into this sprint and add more information to the description.

File: src/browser/api_protocol/api_handlers/system.js [hunk 1]
```diff
@@ -49,6 +49,7 @@ function SystemApiHandler() {
     };
 
     let SystemApiHandlerMap = {
+        'resource-fetch-authenticate': { apiFunc: authenticateResourceFetch },
```
Comment (HarsimranSingh) [inline src/browser/api_protocol/api_handlers/system.js]: Alphabetical order

File: src/browser/authentication_delegate.js [hunk 2]
```diff
@@ -64,7 +66,8 @@ function createAuthUI(identity) {
             defaultCentered: true,
             defaultTop: true,
             frame: false
-        }
+        },
+        runtimeAuthDialog: true // skip checks for shouldCloseRuntime
```
Comment (HarsimranSingh) [inline src/browser/authentication_delegate.js]: Can this be `_runtimeAuthDialog` to show that it's an internal option?

File: src/login/index.html [hunk 3]
```diff
@@ -153,16 +153,30 @@
...
     function cancel() {
         let targetWindow = getTargetWindow();
-
-        targetWindow.authenticate('', '', ()=> fin.desktop.Application.getCurrent().close());
+        const resourceFetch = getParameterByName('resourceFetch');
+        if (resourceFetch) {
+            const uuid = getParameterByName('uuid');
+            fin.desktop.Window.getCurrent().hide(() => { // so error popup is not covered by this window
+                fin.desktop.System.authenticateResourceFetch(uuid, '', '',
+                    () => fin.desktop.Application.getCurrent().close());
```
Comment (datamadic) [inline src/login/index.html]: If we fail here this window will live forever, perhaps handle the error callback as well?

File: src/browser/cached_resource_fetcher.ts [hunk 4]
```diff
@@ -213,10 +228,8 @@ function download(fileUrl: string, filePath: string): Promise<any> {
             });
         });
 
-        request.once('login', (authInfo: any, callback: Function) => {
-            //Simply provide empty credentials to raise the authentication error.
-            //https://github.com/rdepena/electron/blob/master/docs/api/client-request.md#event-login
-            callback();
+        request.on('login', (authInfo: any, callback: Function) => {
```
Comment (datamadic) [inline src/browser/cached_resource_fetcher.ts]: Should we clean up on the 'error' event like [here](https://github.com/HadoukenIO/core/pull/280/files#diff-f2ab184e212da874890dc8d44365e5bcR321)?

File: src/browser/authentication_delegate.js [hunk 5]
```diff
@@ -64,7 +66,8 @@ function createAuthUI(identity) {
             defaultCentered: true,
             defaultTop: true,
             frame: false
-        }
+        },
+        _runtimeAuthDialog: true // skip checks for shouldCloseRuntime
```
Comment (datamadic) [inline src/browser/authentication_delegate.js]: Don't we already have a persistent or non persistent option?

**Pull request merged**